base/errorshow.jl

```diff
@@ -78,7 +82,17 @@ function showerror(io::IO, ex::TypeError)
         else
             ctx = "in $(ex.func), in $(ex.context)"
         end
-        print(io, ctx, ", expected ", ex.expected, ", got ", targs...)
+        if type_param_error
+            print(io, """
+                      values of type $(typeof(ex.got)) are not valid as type parameters. Valid type parameters are
+                          - Types
+                          - Bits values
+                          - Symbols
+                          - Tuples of symbols or bits values
```

Comment:
```suggestion
                          - Tuples of symbols, bits values or other tuples thereof
```

Reply:
It is not recursive:
```
julia> Tuple{((:a,),)}
ERROR: TypeError: in Tuple, in parameter, expected Type, got a value of type Tuple{Tuple{Symbol}}
Stacktrace:
 [1] top-level scope
   @ REPL[2]:1
```

Reply:
Interesting, it does work with ints though:
```julia
julia> Tuple{((1,),)}
Tuple{((1,),)}
```

Reply:
```suggestion
                          - Tuples of symbols, bits values or other tuples of bits values
```

Reply:
That is an isbits value

Reply:
Ah, I see. Since we explicitly list tuples here, perhaps
```suggestion
                          - Tuples of symbols or bits values, including tuples of bits values
```
would clarify this a bit?

Reply:
how about just "- tuples of symbols"? (see the new commit)